@here/harp-datasource-protocol/lib/Theme.ts

```diff
@@ -196,17 +191,34 @@ export interface StylePriority {
 }
 
 /**
- * A type representing HARP themes with all the styleset declarations
- * grouped in one `Array`.
- *
- * @internal This type will merge with {@link Theme}.
+ * Map theme is used to define what features are shown and how the map is styled, for example
+ * which lightning is used or whether fog should be displayed.
  */
-export type FlatTheme = Omit<Theme, "styles"> & {
+export interface Theme extends BaseTheme {
+    /**
+     * Map styles available for datasources used to render the map.
+     */
+    styles?: Styles;
+}
+
+/**
+ * Map theme is used to define what features are shown and how the map is styled, for example
+ * which lightning is used or whether fog should be displayed.
+ */
+export interface FlatTheme extends BaseTheme {
     /**
      * The style rules used to render the map.
      */
     styles?: StyleSet;
-};
+}
+
+/**
+ * Map theme is used to define what features are shown and how the map is styled, for example
+ * which lightning is used or whether fog should be displayed.
+ *
+ * @internal
+ */
+export type ThemeOrFlatTheme = Theme | FlatTheme;
```

Comment:
This is the entry point used to generate a JSON schema supporting both the legacy and the new flat format. In theory we should remove this as soon as we stop supporting LegacyThemes.